### PR TITLE
#74 data model v2

### PR DESCRIPTION
**Pull request by kardan**
A PR for the Brave and true.

### CODE REVIEW

File: backend/dev/user.clj [hunk 1]
```diff
@@ -8,9 +8,11 @@
             [meta-merge.core :refer [meta-merge]]
             [reloaded.repl :refer [system init start stop go reset]]
             [ring.middleware.stacktrace :refer [wrap-stacktrace]]
-            [duct.component.ragtime :as ragtime]
+            ;; [duct.component.ragtime :as ragtime]
+            ;; [org.akvo.dash.component.migra :as migrathor]
```
Comment (jonase) [inline backend/dev/user.clj]: @kardan  Should we remove the commented out code?

File: backend/project.clj [hunk 2]
```diff
@@ -7,31 +7,35 @@
...
+                 [compojure "1.5.0"]
+                 [duct "0.5.10"]
                  [duct/hikaricp-component "0.1.0"]
-                 [duct/ragtime-component "0.1.3"]
                  [environ "1.0.2"]
                  [meta-merge "0.1.1"]
+                 [org.akvo/commons "0.4.4-SNAPSHOT"]
+                 [org.akvo/commons "0.4.4-SNAPSHOT"]
+                 [org.akvo/resumed "0.1.0-SNAPSHOT"]
+                 [org.akvo/resumed "0.1.0-SNAPSHOT"]
```
Comment (jonase) [inline backend/project.clj]: remove the double imports?

File: backend/src/org/akvo/dash/component/tenant_manager.clj [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+(hugsql/def-db-fns "org/akvo/dash/component/tenant_manager.sql")
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Middleware
+;;;
+
+(defn wrap-label-tenant
+  "Parses the first dns label as tenant id and adds it to the request map as
+  tenant-id."
+  {:arglist '([handler])}
```
Comment (jonase) [inline backend/src/org/akvo/dash/component/tenant_manager.clj]: @kardan General question: Why is the `:arglist` needed or what is it used for?

File: backend/src/org/akvo/dash/endpoint/dataset.clj [hunk 4]
```diff
@@ -1,131 +1,97 @@
...
-                (scheduling/schedule #(import/job db
-                                                  (-> res :i :id)))
-                (rr (assoc resp "status" "PENDING")))))
+          (handle-datasets-POST! request
+                                 (connection tm tenant))
           (catch Exception e
-            (rr {:error "Could not complete upload"
-                 :status "FAILED"})))))
+            (pprint e)
+            (pprint (.getNextExcpetion e))
```
Comment (jonase) [inline backend/src/org/akvo/dash/endpoint/dataset.clj]: Are these debug statements that we want to remove?

File: backend/dev/user.clj (hunk 1 above)
Comment (kardan) [inline backend/dev/user.clj]: Yes

File: backend/src/org/akvo/dash/endpoint/dataset.clj (hunk 4 above)
Comment (kardan) [inline backend/src/org/akvo/dash/endpoint/dataset.clj]: Not forever but for a bit.

File: backend/src/org/akvo/dash/component/tenant_manager.clj (hunk 3 above)
Comment (kardan) [inline backend/src/org/akvo/dash/component/tenant_manager.clj]: Let's remove them.

File: backend/project.clj (hunk 2 above)
Comment (kardan) [inline backend/project.clj]: Yes.

File: backend/provision/helpers/tardis.sql [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            RETURN NEW;
+        ELSIF TG_OP = 'UPDATE' THEN
+            OPEN c FOR EXECUTE 'SELECT _validrange FROM history.' || TG_TABLE_NAME ||
+              ' WHERE id = $1 ORDER BY _validrange DESC LIMIT 1 FOR UPDATE'
+              USING NEW.id;
+            FETCH FROM c INTO tt;
+
+            IF isempty(tstzrange(lower(tt), now(), $$[)$$)) THEN
+
+                -- What edge case are we missing by commenting the following lines of code?
```
Comment (iperdomo) [inline backend/provision/helpers/tardis.sql]: @kardan we should remove this commented code, we know what _edge cases_ are we missing

Comment (kardan) [inline backend/provision/helpers/tardis.sql]: Agree, but @iperdomo you want to remove the comment or everything that is commented out?

File: backend/src/org/akvo/dash/endpoint/dataset.clj [hunk 6]
```diff
@@ -1,131 +1,97 @@
...
-               (condp = (:status r)
-                 "OK"      (assoc resp-map
-                                  :columns (json/parse-string
-                                            (slurp (get-in r [:noble "noble"]))))
-                 "FAILURE" (assoc resp-map
-                                  :reason
-                                  "Could not find dataset on remote server (404)")
-                 resp-map)))))) ;; "PENDING" do nothing
+        (fn [{tenant :tenant :as request}]
+          (let [{:keys [:id :d :name :status :created :modified] :as ds}
```
Comment (jonase) [inline backend/src/org/akvo/dash/endpoint/dataset.clj]: @kardan For my own understanding. What is `:d` here?

File: backend/src/org/akvo/dash/import.clj [hunk 7]
```diff
@@ -1,177 +1,75 @@
...
-     :content-type   (get-in resp [:headers "Content-Type"])
-     :digest         (sha1 data)
-     :file-extension (last (str/split url #"\."))}))
+(defmulti parse-content
+  "Hook that enables data type special cases."
+  (fn [content] (:type content))
+  :default
+  (fn [content]
+    (throw (IllegalArgumentException.
+            (str "Can't process dataa of " (:type content) " type.")))))
```
Comment (jonase) [inline backend/src/org/akvo/dash/import.clj]: @kardan typo "dataa"

File: backend/src/org/akvo/dash/import.sql [hunk 8]
```diff
@@ -54,4 +24,4 @@ VALUES (:id, :dataset-id, :type, :c-name, :c-order)
 -- :name insert-dataset-columns :! n
 -- :doc Insert new dataset columns with *tuple parameters
 INSERT INTO dataset_column(id, dataset_id, type, name, c_name, c_order)
-VALUES :tuple*:columns
+VALUES :tuple*:columnsq
```
Comment (jonase) [inline backend/src/org/akvo/dash/import.sql]: @kardan is "columnsq" a typo?

File: backend/provision/helpers/tardis.sql (hunk 5 above)
Comment (iperdomo) [inline backend/provision/helpers/tardis.sql]: @kardan everything commented out

File: backend/src/org/akvo/dash/endpoint/dataset.clj (hunk 6 above)
Comment (kardan) [inline backend/src/org/akvo/dash/endpoint/dataset.clj]: The dataset "data". I'm guessing all this will go away with the new dataset implementation.

**Pull request merged**